frontend/src/components/ViewActivities/activityfns.js

```diff
@@ -0,0 +1,75 @@
+import * as DBFIELDS from '../../constants/dbconstants';
+import app from '../Firebase';
+
+const db = app.firestore();
+
+/**
+ * Gets the list of activities from the server. 
+ * @param {string} tripId The trip ID.
+ */
+export async function getActivityList(tripId) {
+  return new Promise(function(resolve, reject) {
+    let tripActivities = [];
+    
+    db.collection(DBFIELDS.COLLECTION_TRIPS).doc(tripId)
+    .collection(DBFIELDS.COLLECTION_ACTIVITIES).get()
+    .then(querySnapshot => {
+      querySnapshot.forEach(doc => {
```

Comment:
+4 spaces for continuation lines, then +2 spaces for each block.
```suggestion
    db.collection(DBFIELDS.COLLECTION_TRIPS).doc(tripId)
        .collection(DBFIELDS.COLLECTION_ACTIVITIES).get()
        .then(querySnapshot => {
          querySnapshot.forEach(doc => {
```